lazer/sdk/rust/client/src/client.rs

```diff
@@ -0,0 +1,113 @@
+use std::time::Duration;
+
+use crate::{
+    resilient_ws_connection::PythLazerResilientWSConnection, ws_connection::AnyResponse,
+    CHANNEL_CAPACITY,
+};
+use anyhow::{bail, Result};
+use backoff::ExponentialBackoff;
+use futures_util::stream;
+use pyth_lazer_protocol::subscription::{SubscribeRequest, SubscriptionId};
+use tokio::sync::mpsc::{self, error::TrySendError};
+use tokio_stream::{wrappers::ReceiverStream, StreamExt};
+use tracing::{error, warn};
+use ttl_cache::TtlCache;
+
+const DEDUP_CACHE_SIZE: usize = 100_000;
+const DEDUP_TTL: Duration = Duration::from_secs(10);
+
+pub struct PythLazerClient {
+    endpoints: Vec<String>,
+    access_token: String,
+    num_connections: usize,
+    ws_connections: Vec<PythLazerResilientWSConnection>,
+    receivers: Vec<mpsc::Receiver<AnyResponse>>,
+    backoff: ExponentialBackoff,
+}
+
+impl PythLazerClient {
+    /// Creates a new client instance
+    ///
+    /// # Arguments
+    /// * `endpoints` - A vector of endpoint URLs
+    /// * `access_token` - The access token for authentication
+    /// * `num_connections` - The number of WebSocket connections to maintain
+    pub fn new(
+        endpoints: Vec<String>,
+        access_token: String,
+        num_connections: usize,
+        backoff: ExponentialBackoff,
```

Comment:
this seems to be part of your public interface. you need to reexport it to downstream consumers. Other thing is to have a wrapper that has what you need (because i see you are not supporting everything there)

Reply:
added a builder wrapper.

Reply:
Done.